### PR TITLE
Feature/d3asim 3307 Created an interface for the results classes. Added a manager class for all the simulation result objects. 

### PR DESCRIPTION
**Pull request by spyrostz**
Potential improvement: currently the unit tests for the results classes are located either in d3a or in d3a-web, one more improvement to the process would be to port these to the d3a-interface, so that the code is tested in the same repo that it is created.

### CODE REVIEW

File: d3a_interface/sim_results/export_unmatched_loads.py [hunk 1]
```diff
@@ -17,9 +17,11 @@
 """
```
Comment (hannesdiedrich) [inline d3a_interface/sim_results/export_unmatched_loads.py:17]: We should really deal with D3ASIM-3324 soon!

**Review by hannesdiedrich — DISMISSED** [review]
Great! Thank you! LGTM

File: d3a_interface/sim_results/results_abc.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+from typing import Dict, List
+
+
+class ResultsBaseClass(ABC):
+
+    @staticmethod
+    def _validate_update_parameters(area_result_dict, core_stats, current_market_slot):
```
Comment (faizan2590) [inline d3a_interface/sim_results/results_abc.py:8]:
```suggestion
    def _has_update_parameters(area_result_dict, core_stats, current_market_slot):
```
Since it expects to return boolean `True or False` rather than itself raising any exception.

**Review by faizan2590 — COMMENTED** [review]
one comment, other than that LGTM.

File: d3a_interface/sim_results/results_abc.py (hunk 2 above)
Comment (spyrostz) [reply d3a_interface/sim_results/results_abc.py:8]: Why not, I'll adapt it, thanks!

File: d3a_interface/sim_results/energy_trade_profile.py [hunk 3]
```diff
@@ -145,3 +151,44 @@ def convert_timestamp_strings_to_datetimes(profile):
...
+
+    def restore_area_results_state(self, area_uuid, last_known_state_data):
+        pass
+
+    @property
+    def plot_results(self):
+        return self.convert_timestamp_strings_to_datetimes(
+            convert_pendulum_to_str_in_dict(
+                self.traded_energy_profile, {}, ui_format=True)
+        )
```
Comment (hannesdiedrich) [inline d3a_interface/sim_results/energy_trade_profile.py:184]: Sorry, I know this has been in the d3a for a while, but why do we first have to convert first to string and then to datetimes? Isn't the `plot_energy_profile` able to handle pendulum objects?

File: d3a_interface/sim_results/area_throughput_stats.py [hunk 4]
```diff
@@ -87,3 +90,26 @@ def update_results(self, area_dict, core_stats, current_market_time_slot_str):
         for child in area_dict['children']:
             if child['type'] == "Area":
                 self.update_results(child, core_stats, current_market_time_slot_str)
+
+    @staticmethod
+    def merge_results_to_global(market_trade: Dict, global_trade: Dict, *_):
```
Comment (hannesdiedrich) [inline d3a_interface/sim_results/area_throughput_stats.py:95]: AFAIU, `*_` is a way to ignore all additional arguments, right?
Why is this needed here?

**Review by hannesdiedrich — DISMISSED** [review]
Two comments or rather questions, LGTM

File: d3a_interface/sim_results/area_throughput_stats.py (hunk 4 above)
Comment (spyrostz) [reply d3a_interface/sim_results/area_throughput_stats.py:95]: Because `slot_list` from the abc is required only in some implementations of the `merge_results_to_global` method. Therefore if I created a method with only 2 arguments, Python would complain that the abstractmethod does not adhere to the same number of arguments as the base class method.

File: d3a_interface/sim_results/energy_trade_profile.py (hunk 3 above)
Comment (spyrostz) [reply d3a_interface/sim_results/energy_trade_profile.py:184]: Good question! I would also love to remove this, let me dig a bit in d3a export.py and I'll let you know what I find

**Review by faizan2590 — DISMISSED** [review]
LGTM,  though one old [comment](https://github.com/gridsingularity/d3a-interface/pull/167#discussion_r581066910) from my side to be addressed

File: d3a_interface/sim_results/energy_trade_profile.py (hunk 3 above)
Comment (spyrostz) [reply d3a_interface/sim_results/energy_trade_profile.py]: Thanks for the suggestion, I removed it and also added the call of the `add_sold_bought_lists` to this method too, in order to avoid the direct import of the class in the d3a code.
